.github/workflows/ci.yml

```
@@ -47,13 +51,20 @@ jobs:
         ./cc-test-reporter before-build
       env:
         CC_TEST_REPORTER_ID: aff2c7b9e07e54d5fc9e5588d2e2a8bab4f69950d35000edc2b6250bbaba477d
-    - name: Build and test with Rake
+    - name: Build and test with Rake (ruby-2.x)
+      if: startsWith(matrix.ruby_version, '2')
       run: |
         gem install bundler:1.17.3
         bundle _1.17.3_ update
         bundle _1.17.3_ install --gemfile spec/gemfiles/${{ matrix.gemfile }} --jobs 4 --retry 3
         bundle _1.17.3_ exec rake code_analysis
         bundle _1.17.3_ exec rspec
+    - name: Build and test with Rake (ruby-3.x)
+      if: startsWith(matrix.ruby_version, '3')
+      run: |
+        bundle install --gemfile spec/gemfiles/${{ matrix.gemfile }} --jobs 4 --retry 3
+        bundle exec rake code_analysis
+        bundle exec rspec
```

```suggestion
    - name: Build and test
      run: |
        bundle install --gemfile spec/gemfiles/${{ matrix.gemfile }} --jobs 4 --retry 3
        bundle exec rake code_analysis
        bundle exec rspec
```

why can't we do just that? why do we even need to have bundler `1.17.3` ?

---

That's how it was originally.

I believe we used 1.17.3 because otherwise it didn't work for older rubies, maybe for 2.3 and 2.4, but I don't remember now. I could test using only `bundle exec` without the version, but maybe in another PR because this already introduces a failure and is going to be hard to determine the issues if any.